lib/manager/sbt/readme.md

```diff
@@ -0,0 +1,245 @@
+
+Scala syntax
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+I think reading the files with Scala would be the easier. Due to the fact that the files are plain Scala code, the
+dependencies can be declared in multiple ways.
+
+But, I guess there is two most commonly used way to declare dependencies and using regex could work for these formats.
+
+Another option might be to use ScalaJS. I don't know it but it has same syntax as Scala and can be read/used within Javascript code.
+
+If we can use Scala or ScalaJS, the idea would be to execute the `build.sbt` or `plugins.sbt` file and work on the `libraryDependencies` or `dependencyOverrides` variables.
+
+_Update (thanks to @ChristianMurphy)_: https://github.com/scalameta/scalameta running on http://www.scala-js.org could be a way to parse Scala without depending on a JDK.
+E.G. https://astexplorer.net/#/gist/027dce11e6927b4ad39ea097ce6289b9/ea4d1048f32063a71d727d01178ae6d01087a62f
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+Dependencies can have a "scope" like `Test` or `Provided`.
+
+Renovate shouldn't care of scopes: they should be kept as is but they don't have any impact on the "new dependency resolution".
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+##### build.sbt
+
+Most common syntaxes:
+
+```scala
+// For Scala dependency (notice the double %%)
+libraryDependencies += "com.typesafe.play" %% "play-ws" % "2.6.10"
+// For Java dependency
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18"
+
+// With a scope
+libraryDependencies += "org.mockito" % "mockito-all" % "1.10.18" % Test
+
+// Multiple dependencies at the same time
+libraryDependencies ++= Seq(
+  "com.typesafe.play" %% "play-ws" % "2.6.10",
+  "org.mockito" % "mockito-all" % "1.10.18"
+)
+
+// Defining some dependency overriding (same as for build.sbt)
+dependencyOverrides += "com.google.guava" % "guava" % "23.0"
+// Multiple at once
+dependencyOverrides ++= Seq(
+  "org.scala-lang.modules" %% "scala-parser-combinators" % "1.1.0",
+  "com.google.guava" % "guava" % "23.0"
+)
+```
+
+There is two variables to look for `libraryDependencies` and `dependencyOverrides`.
+
+More custom syntaxes with plain Scala code:
+
+```scala
+val myDependencyVersion = "2.6.10"
+val myDependencyGroupId = "com.typesafe.play"
+val myDependencyArtifactId = "play-json"
+val myDependency: ModuleID = myDependencyGroupId %% myDependencyArtifactId % myDependencyVersion
+
+libraryDependencies += myDependency
+```
+
+You could imagine some `if` statements as well as any other Scala syntax but I think it's pretty rare.
+
+##### plugins.sbt
+
+Most common syntaxes:
+
+```scala
+// Declaring a plugin
+addSbtPlugin("com.eed3si9n" % "sbt-buildinfo" % "0.9.0")
```

Comment:
Are these single `%` declarations therefore java plugins, or do scala plugins use single % ?

Reply:
Actually `%` doesn't mean the dependency is Java. It's just that `%%` is a shortcut for Scala libraries that allows to omit the Scala version (otherwise it would be defined as `groupId % artifactId_scalaVersion % version`). I updated the file with more details about this.

Regarding plugins, I believe they are not dependent on a Scala version and that's why they are described with single `%`. Plus I don't think you can have Java plugins, sbt plugins are necessarily written in Scala.